src/halse_se05x.c

```diff
@@ -484,7 +496,9 @@ static int halse_se05x_warm_reset_dev(struct halse_se05x_dev *dev)
 		return -1;
 	}
 
-	free(dev->atr);
+	if (dev->atr)
+		free(dev->atr);
+
```

Comment:
man 3 free says:
       The free() function frees the memory space pointed  to  by  ptr,  which
       must  have  been  returned by a previous call to malloc(), calloc(), or
       realloc().  Otherwise, or if free(ptr) has already been called  before,
       undefined behavior occurs.  If ptr is NULL, no operation is performed.